src/server.py

```diff
@@ -3,39 +3,121 @@
 Licensed under the MIT (Expat) License. See the LICENSE file found in the
 top-level directory of this distribution.
 '''
-from sanic import Sanic, response
-from sanic.response import json
+import aiohttp
 
 from json import dumps
+from decouple import config
+
+from sanic import Sanic, response
+from sanic.response import json, text
+
+from sanic_oauth.blueprint import oauth_blueprint, login_required
+
+from sanic_session import InMemorySessionInterface
 
-from listeners import listener_blueprint
+from initialise_db import initdb_blueprint
+from authentication import auth_blueprint
 
 
 app = Sanic(__name__)
-app.blueprint(listener_blueprint)
 
-table = listener_blueprint.table
+app.blueprint(initdb_blueprint)
+app.blueprint(oauth_blueprint)
+app.blueprint(auth_blueprint)
+
+table = initdb_blueprint.table
+
+# ----------------------------------------------------------------------------
+# AUTHENTICATION
+# ----------------------------------------------------------------------------
+
+app.session_interface = InMemorySessionInterface()
+app.config.OAUTH_PROVIDER = config('OAUTH_PROVIDER')
+app.config.OAUTH_REDIRECT_URI = config('OAUTH_REDIRECT_URI')
+app.config.OAUTH_SCOPE = config('OAUTH_SCOPE')
+app.config.OAUTH_CLIENT_ID = config('OAUTH_CLIENT_ID')
+app.config.OAUTH_CLIENT_SECRET = config('OAUTH_CLIENT_SECRET')
+
+
+@app.middleware('request')
+async def add_session_to_request(request):
+    await request.app.session_interface.open(request)
+
+
+@app.middleware('response')
+async def save_session(request, response):
+    '''
+    See SESSION_ERROR in Documentation.
+    '''
+    try:
+        user_info = request['session']['user_info']
+        try:
+            request['session']['user_info'] = dict(
+                [(attr, getattr(user_info, attr))
+                  for attr in user_info.default_attrs]
+            )
+        except AttributeError:
+            pass
```

Comment:
I understand that this works universally. If you are fine with this, I'm fine too. If you want less magic here, looking at what the library fetches from Google, this should work as well:
```suggestion
        request['session']['user_info'] = {
            attr: getattr(user_info, attr)
              for attr in {'id', 'email', 'verified_email', 'first_name', 'last_name', 'picture'}
        )
```
or taking into account how you use the `user` object, this should work too:
```suggestion
        request['session']['user_info'] = {'email': user_info.email}
```
But as I said... If you are fine with your solution, I'm fine too...

Reply:
I was going to use the last example, but I won't know which attributes will be needed until the app's complete (user id is being used as per the last commit for example). It's commented out in the code for the future.